### PR TITLE
Added scores caching

### PR DESCRIPTION
**Pull request by Dantemss**
Still needs invalidations when updating the task_plan etc

### CODE REVIEW

File: db/background_migrate/20200617142820_cache_points_for_existing_tasks.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
+class CachePointsForExistingTasks < ActiveRecord::Migration[5.2]
+  def up
```
[the remark this answers is not in the document]
Comment (Dantemss) [reply db/background_migrate/20200617142820_cache_points_for_existing_tasks.rb:2]: OK

File: app/controllers/api/v1/tasking_plans_controller.rb [hunk 2]
```diff
@@ -16,33 +16,43 @@ class Api::V1::TaskingPlansController < Api::V1::ApiController
...
-    Tasks::Models::TaskedExercise.joins(:task_step).where(
-      task_step: { tasks_task_id: tasks.map(&:id) }
-    ).update_all '"published_points" = "grader_points", "published_comments" = "grader_comments"'
+      tasks.update_all grades_last_published_at: Time.current
 
-    tasks.update_all grades_last_published_at: Time.current
+      queue = task_plan.is_preview ? :preview : :dashboard
+      Tasks::UpdateTaskCaches.set(queue: queue).perform_later(
+        task_ids: tasks.map(&:id), update_cached_attributes: true, queue: queue.to_s
+      )
```
Comment (nathanstitt) [inline app/controllers/api/v1/tasking_plans_controller.rb:50]: Seems like the above updates need to be inside a transaction, or moved to a routine where the transaction would be automatic

**Comment by Dantemss** [comment]
https://travis-ci.org/github/openstax/tutor-server/builds/699789720

File: app/controllers/api/v1/tasking_plans_controller.rb (hunk 2 above)
Comment (Dantemss) [reply app/controllers/api/v1/tasking_plans_controller.rb:50]: Hm I did add a transaction though.

**Comment by Dantemss** [comment]
Build is failing...

File: app/controllers/api/v1/tasking_plans_controller.rb (hunk 2 above)
Comment (nathanstitt) [reply app/controllers/api/v1/tasking_plans_controller.rb:50]: Grr, I'd missed the `::Tasks::Models::TaskingPlan.transaction do` at the top of the method.  Sorry!

**Pull request merged**